### ISSUE TITLE
First feedback

### ISSUE THREAD

**Issue by Eddie85** [arduino-libraries/ArduinoMqttClient#1]
Dear Sandeep

I'm glad I found a promising solution. Since this point it is far better than the other solutions.
Already did some tests on different devices (Arduino Nano + W5100 & NodeMCU Esp32S) and recognized some bugs and defect.

My MQTT broker is ActiveMQ and when my device subscribed and received a message somehow ACK didn't reached the broker, amount of dequeued messages wasn't change on GUI but on Serial out I saw the messages.

By the way, is it possible to use this library as **"Request/Reply"** solution? Bombarded my devices with ~1000messages ,- ack back to broker also missed.  And I could not send an answer in onMessage function (after the 3rd message Arduino crashed, maybe because of memory limitations.) But when I moved beginMessage; print;endMessage into the main loop then only the last message 
was answered because poll function didn't turned back after a received message.

So basically is it possible to pre-configure the amount of handled message via ::poll ? Receive one message, return to main loop, answer it if want, etc... 
Is it on plan? 

Br, Ed

**Comment by sandeepmistry** [comment]
Hi @Eddie85,

Could you please provide a minimal sketch to reproduce the issue and give some more details on the message that is being received (topic and payload).

Thanks.

**Comment by sandeepmistry** [comment]
There was an issue with the ACK using the wrong packet id, this has been corrected in https://github.com/arduino-libraries/ArduinoMqttClient/commit/253a12e188c521712e2547b252ebf15d659f2b0f. Please let us know if you still are facing issues, the issue can then be re-opened.

This change will be available in the next 0.1.1 release of the library.

**Comment by Eddie85** [comment]
Hi Sandee

Thanks, I'll test it at the weekend.
Back to the other issue when poll wasnt returning after 1 message? It consumed all of them and then onmessage funtion returned and main loop was processing?
